pyproject.toml

```diff
@@ -56,24 +54,15 @@ importlib-metadata = { version = "^1.0", python = "<3.8" }
 protobuf = "^3.12.1"
 dataclasses = { version = "==0.6", markers = "python_version < '3.7'"  }
 # Optional dependencies
-tensorflow-cpu = { version = "==2.6.2", optional = true }
-torch = { version = "^1.10.1", optional = true }
-torchvision = { version = "^0.11.2", optional = true }
-boto3 = { version = "^1.12.36", optional = true }
-boto3_type_annotations = { version = "^0.3.1", optional = true }
-paramiko = { version = "^2.7.1", optional = true }
-docker = { version = "^4.2.0", optional = true }
-matplotlib = { version = "^3.2.1", optional = true }
-tqdm = { version = "^4.48.2", optional = true }
 ray = { extras = ["default"], version = "==1.6.0", optional = true }
 
 [tool.poetry.extras]
 simulation = ["ray"]
-baseline = ["tensorflow-cpu", "boto3", "boto3_type_annotations", "paramiko", "docker", "matplotlib"]
-examples-pytorch = ["torch", "torchvision", "tqdm"]
-examples-tensorflow = ["tensorflow-cpu"]
-http-logger = ["boto3", "boto3_type_annotations", "tensorflow-cpu", "boto3", "boto3_type_annotations", "paramiko", "docker", "matplotlib"]
-ops = ["boto3", "boto3_type_annotations", "paramiko", "docker"]
+baseline = []
+examples-pytorch = []
+examples-tensorflow = []
+http-logger = []
+ops = []
```

Comment:
I think these can also be removed

Reply:
They could be removed, but it'd be a breaking change. Keeping them as an empty list makes the extras a no-op, they can be completely removed in a later release.

Reply:
Could you explain? Removing the packages and dependencies seems to be a breaking change in itself.